core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/io/writeJson.kt

```diff
@@ -261,45 +293,48 @@ internal fun AnyFrame.extractArrayColumn(): DataColumn<*>? {
         }
 }
 
-internal fun KlaxonJson.encodeFrame(frame: AnyFrame): JsonArray<*> {
+internal fun encodeFrame(frame: AnyFrame): JsonArray {
     val valueColumn = frame.extractValueColumn()
     val arrayColumn = frame.extractArrayColumn()
 
     val arraysAreFrames = arrayColumn?.kind() == ColumnKind.Frame
 
     val data = frame.indices().map { rowIndex ->
-        valueColumn
-            ?.get(rowIndex)
-            ?: arrayColumn?.get(rowIndex)
-                ?.let {
-                    if (arraysAreFrames) encodeFrame(it as AnyFrame) else null
-                }
-            ?: encodeRow(frame, rowIndex)
+        valueColumn?.get(rowIndex) ?: arrayColumn?.get(rowIndex)?.let {
```

Comment:
This ?: chain is very hard to understand. I know you just reformatted it, but maybe you know a better way to write it? :)

Reply:
I rewrote it using the `when` construct. it seems more readable now. Could you please take a look?